### PR TITLE
Cleanup migrated AWS resources

### PR DESCRIPTION
**Pull request by aliscott**
* Makes non-usage and non-optional attributes non-pointers, e.g. `*r.Region` -> `r.Region`
* Moves functions into struct methods, e.g. `func httpCostComponent()` → `func (r *APIGatewayV2API) httpCostComponent()`
* Fixes casing of some names, e.g. `DmsReplicationInstance` → `DMSReplicationInstance`
* Fixes some generated attribute names, e.g. `Environment0ComputeType` -> `ComputeType`
* Fixes some spacing

### CODE REVIEW

File: internal/providers/terraform/aws/acm_certificate.go [hunk 1]
```diff
@@ -12,10 +12,12 @@ func getACMCertificate() *schema.RegistryItem {
 	}
 }
 func NewAcmCertificate(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
-	r := &aws.AcmCertificate{Address: strPtr(d.Address), Region: strPtr(d.Get("region").String())}
-	if !d.IsEmpty("certificate_authority_arn") {
-		r.CertificateAuthorityArn = strPtr(d.Get("certificate_authority_arn").String())
+	r := &aws.AcmCertificate{
```
Comment (aliscott) [inline internal/providers/terraform/aws/acm_certificate.go:15]: Should be `ACMCertificate`

File: internal/providers/terraform/aws/acmpca_certificate_authority.go [hunk 2]
```diff
@@ -12,7 +12,11 @@ func getACMPCACertificateAuthorityRegistryItem() *schema.RegistryItem {
 	}
 }
 func NewAcmpcaCertificateAuthority(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
-	r := &aws.AcmpcaCertificateAuthority{Address: strPtr(d.Address), Region: strPtr(d.Get("region").String())}
+	r := &aws.AcmpcaCertificateAuthority{
```
Comment (aliscott) [inline internal/providers/terraform/aws/acmpca_certificate_authority.go:15]: Should be `ACMPCACertificateAuthority`

**Comment by aliscott** [comment]
@sinabakh can you please review this if you get a chance?

**Comment by sinabakh** [comment]
@aliscott LGTM

**Pull request merged**